File: python_ToSXs/python/06-thaphinh/22.py
height = 7
shape = ""
shapeTop = ""
shapeBottom = ""
character = "#"
center = int(height / 2)

line = character * height
for i in range(2, center + 1):
    record = (character + " " * (i - 2) +
              character + " " * (height - 2 * i) +
              character + " " * (i - 2) +
              character)
    shapeTop += record + "\n"

centerCharacter = character + " " * (center - 1) + character + " " * (center - 1) + character + "\n"
for i in range(2 + center, height):
    record = (character + " " * (height - i - 1) +
              character + " " * (2 * i - height - 2) +
              character + " " * (height - i - 1) +
              character)
    shapeBottom += record + "\n"

shape = line + "\n" + shapeTop + centerCharacter + shapeBottom + line
print(shape)

File: cpp_e9zMZ/cpp/06-thaphinh/22.cpp
#include <iostream>

#include "zvn.h"
using namespace std;
int main() {
    int height = 5;
    string shape = "";
    string character = "#";
    string space = " ";
    string line = myRepeat(character, height);

    for (int i = 2; i < height; i++) {
        string record = "";
        for (int j = 1; j <= height; j++) {
            if (j == 1 || j == i || j == height - i + 1 || j == height) {
                record += character;
            } else {
                record += space;
            }
        }
        shape += record + "\n";
    }

    shape = line + "\n" + shape + line;
    cout << shape << endl;
}